packages/dd-trace/src/debugger/devtools_client/remote_config.js

```diff
@@ -0,0 +1,145 @@
+'use strict'
+
+const { workerData: { rcPort } } = require('node:worker_threads')
+const mutexify = require('mutexify/promise')
+const { getScript, probes, breakpoints } = require('./state')
+const session = require('./session')
+const { ackReceived, ackInstalled, ackError } = require('./status')
+const log = require('../../log')
+
+const lock = mutexify()
+let sessionStarted = false
+
+// Example log line probe (simplified):
+// {
+//   id: '100c9a5c-45ad-49dc-818b-c570d31e11d1',
+//   version: 0,
+//   type: 'LOG_PROBE',
+//   where: { sourceFile: 'index.js', lines: ['25'] }, // only use first array element
+//   template: 'Hello World 2',
+//   segments: [...],
+//   captureSnapshot: true,
+//   capture: { maxReferenceDepth: 1 },
+//   sampling: { snapshotsPerSecond: 1 },
+//   evaluateAt: 'EXIT' // only used for method probes
+// }
+//
+// Example log method probe (simplified):
+// {
+//   id: 'd692ee6d-5734-4df7-9d86-e3bc6449cc8c',
+//   version: 0,
+//   type: 'LOG_PROBE',
+//   where: { typeName: 'index.js', methodName: 'handlerA' },
+//   template: 'Executed index.js.handlerA, it took {@duration}ms',
+//   segments: [...],
+//   captureSnapshot: false,
+//   capture: { maxReferenceDepth: 3 },
+//   sampling: { snapshotsPerSecond: 5000 },
+//   evaluateAt: 'EXIT' // only used for method probes
+// }
+rcPort.on('message', async ({ action, conf: probe, ackId }) => {
+  try {
+    await processMsg(action, probe)
+    rcPort.postMessage({ ackId })
+  } catch (err) {
+    rcPort.postMessage({ ackId, error: err })
+    ackError(err, probe)
+  }
+})
+
+async function start () {
+  sessionStarted = true
+  await session.post('Debugger.enable')
+}
+
+async function stop () {
+  sessionStarted = false
+  await session.post('Debugger.disable')
+}
+
+async function processMsg (action, probe) {
+  log.debug(`Received request to ${action} ${probe.type} probe (id: ${probe.id}, version: ${probe.version})`)
+
+  if (action !== 'unapply') ackReceived(probe)
+
+  if (probe.type !== 'LOG_PROBE') {
+    throw new Error(`Unsupported probe type: ${probe.type} (id: ${probe.id}, version: ${probe.version})`)
+  }
+  if (!probe.where.sourceFile && !probe.where.lines) {
+    throw new Error(
+      // eslint-disable-next-line max-len
+      `Unsupported probe insertion point! Only line-based probes are supported (id: ${probe.id}, version: ${probe.version})`
+    )
+  }
+
+  const release = await lock()
```

Comment:
> It's worth leaving a comment as to what this lock is for. I think it's because you want to guarantee some ordering? 

It's to fix a race condition discussed here: https://github.com/DataDog/dd-trace-js/pull/4492#discussion_r1750071683

I'll add a comment 👍 

> At any rate, you don't really need a lock for the default/error case.

You're right, but I thought the code looked cleaner if I just wrapped the entire switch statement in the lock instead of having to set and release the lock inside each `case` block.

Also, I missed adding a `try { ... } finally { release() }` block around the switch statement below to ensure that the lock gets released if anything in-between threw. I'm going to fix that.

But please let me know if you think it could be done prettier / better in another way ☺️

@Qard Do you still think a queue would be better based on what I just wrote?

Reply:
I think this is easy enough to do without a library that we should do it without. Adding libraries, especially ones that are unmaintained like this one, can add arrays of problems and also impact the whole library and not just DI.

Reply:
@rochdev I can implement the lock manually like so:

```js
async function lock () {
  if (lock.p) await lock.p
  let resolve
  lock.p = new Promise((_resolve) => { resolve = _resolve }).then(() => { lock.p = null })
  return resolve
}
```

But there's something to be said about well tested small npm modules to encapsulate complexity and reduce risk of bugs. If I read the code I just posted above, I couldn't say off the top of my head if it would work as expected. I would have to test it, which is what the npm module already does for me.

It's a balance of course. We don't want a million dependencies, but we also don't want to reinvent the well if reinventing comes with added risk (unforeseen bugs in this case).

Btw, to my knowledge @mafintosh is still maintaining `mutexify`. It's just done, unless there's a bug to be fixed - hence why it hasn't received many updates over the last years.

However, this isn't a hill I'm willing to die on 😄 So I can update the code as you suggested, but I'd love to hear if @bengl or @Qard had some input first?

Reply:
Yeah, I would go for a task queue instead. I think it's clearer that way what is intended. Could be as simple as:

```js
let lastTask = Promise.resolve()
function queueTask(taskFn) {
  return (lastTask = last.then(taskFn))
}
```

Reply:
![image](https://github.com/user-attachments/assets/e2b4cebe-9164-41f2-90ca-4acdd817559d)

Reply:
@Qard I don't think the code your posted handles errors that might occur inside `taskFn` 🤔 At least, after implementing it, it looks like errors are escaping the outer try-catch block. I've spent a few hours on it now trying to get it to work properly in all edge cases (which is exactly my point about using an existing npm module 😉)

Reply:
Btw, I added a new commit that contains an test which fails if the race condition is triggered